docs/components/plugins/AppConfigSchemaPropertiesTable.test.js

```diff
@@ -0,0 +1,171 @@
+        },
+      },
+    },
+  },
+  intentFilters: {
+    description: 'Configuration for setting an array of custom intent filters in Android manifest.',
+    example: [
+      {
+        autoVerify: true,
+        data: {
+          host: '*.expo.io',
+        },
+      },
+    ],
+    exampleString: '\n [{ \n "autoVerify": true, \n "data": {"host": "*.expo.io" \n } \n }]',
+    type: 'array',
+    uniqueItems: true,
+    items: {
+      type: 'object',
+      properties: {
+        autoVerify: {
+          description:
+            'You may also use an intent filter to set your app as the default handler for links',
+          type: 'boolean',
+        },
+        data: {
+          type: ['array', 'object'],
+          items: {
+            type: 'object',
+            properties: {
+              host: { description: 'the host, e.g. `myapp.io`', type: 'string' },
+            },
+            additionalProperties: false,
+          },
+          properties: {
+            host: { type: 'string' },
+          },
+          additionalProperties: false,
+        },
+      },
+      additionalProperties: false,
+      required: ['action'],
+    },
+  },
+};
+
+test('correct property and subproperty indent styling', () => {
+  const { getByTestId } = render(
+    <SluggerContext.Provider value={new GithubSlugger()}>
+      <AppConfigSchemaPropertiesTable schema={testSchema} />
+    </SluggerContext.Provider>
+  );
+
+  const level_0_style = `
+    marginLeft: 12px,
+    display: 'block',
+    listStyleType: 'circle',
+    width: 'fit-content',
+    overflowX: 'visible',
+  `;
+
+  const level_1_style = `
+    marginLeft: 44px,
+    display: 'inline',
+    listStyleType: 'default',
+    width: 'fit-content',
+    overflowX: 'visible',
+  `;
+
+  const level_2_style = `
+    marginLeft: 76px,
+    display: 'inline',
+    listStyleType: 'circle',
+    width: 'fit-content',
+    overflowX: 'visible',
+  `;
+
+  //level-0 property
+  //the #### is added to link this property name by markdown and included in the data-testid
+  expect(getByTestId('#### `name`')).toHaveStyle(level_0_style);
+
+  //level-0 property
+  expect(getByTestId('#### `androidNavigationBar`')).toHaveStyle(level_0_style);
+
+  //level-1 subproperty
+  expect(getByTestId('`visible`')).toHaveStyle(level_1_style);
+
+  //level-2 subproperty
+  expect(getByTestId('`always`')).toHaveStyle(level_2_style);
+
+  //level-1 subproperty
+  expect(getByTestId('`backgroundColor`')).toHaveStyle(level_1_style);
+
+  //level-0 property
+  expect(getByTestId('#### `intentFilters`')).toHaveStyle(level_0_style);
+
+  //level-1 subproperty
+  expect(getByTestId('`autoVerify`')).toHaveStyle(level_1_style);
+
+  //level-1 subproperty
+  expect(getByTestId('`data`')).toHaveStyle(level_1_style);
+
+  //level-2 subproperty
+  expect(getByTestId('`host`')).toHaveStyle(level_2_style);
+});
+
+test('correct description add-ons (bareWorkflow, regexHuman, etc.)', () => {
```

Comment:
it may be easier to test `appendProperty` and `createDescription` directly - this would make it easier to work with this code in the future because they seem like the parts that we're most likely to change